StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1009UnitTests.cs

```diff
@@ -0,0 +1,557 @@
+        {
+            var validStatement = @"new System.Action(() => { })();";
+
+            await this.TestWhitespaceInStatementOrDeclAsync(validStatement, EmptyDiagnosticResults).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSpaceClosingParenthesisFollowedByParenthesisPairAsync()
+        {
+            var invalidStatement = @"new System.Action(() => { }) ();";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(" not", "followed").WithLocation(7, 40);
+
+            await this.TestWhitespaceInStatementOrDeclAsync(invalidStatement, expected).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestNoSpaceParenthesisFollowedByBracketAsync()
+        {
+            var validStatement = @"var a = GetA()[0];";
+
+            await this.TestWhitespaceInStatementOrDeclAsync(validStatement, EmptyDiagnosticResults).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSpaceParenthesisFollowedByBracketAsync()
+        {
+            var invalidStatement = @"var a = GetA() [0];";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(" not", "followed").WithLocation(7, 26);
+
+            await this.TestWhitespaceInStatementOrDeclAsync(invalidStatement, expected).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSpaceParenthesisFollowedByColonAsync()
+        {
+            var validStatement = @"var x = true ? GetA() : GetB();";
+
+            await this.TestWhitespaceInStatementOrDeclAsync(validStatement, EmptyDiagnosticResults).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestNoSpaceParenthesisFollowedByColonAsync()
+        {
+            var invalidStatement = @"var x = true ? GetA(): GetB();";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(string.Empty, "followed").WithLocation(7, 33);
+
+            await this.TestWhitespaceInStatementOrDeclAsync(invalidStatement, expected).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSpaceParenthesisFollowedByQuestionAsync()
+        {
+            var validStatement = @"var x = (true == true) ? GetA() : GetB();";
+
+            await this.TestWhitespaceInStatementOrDeclAsync(validStatement, EmptyDiagnosticResults).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestNoSpaceParenthesisFollowedByQuestionAsync()
+        {
+            var invalidStatement = @"var x = (true == true)? GetA() : GetB();";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(string.Empty, "followed").WithLocation(7, 34);
+
+            await this.TestWhitespaceInStatementOrDeclAsync(invalidStatement, expected).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        protected override IEnumerable<DiagnosticAnalyzer> GetCSharpDiagnosticAnalyzers()
```

Comment:
By convention, yes, but *technically* we violate the ordering rules everywhere else. StyleCop says private members should appear after protected members.